File: z-clients/javascript/src/models/serverState.ts
// this file is @generated

export enum ServerState {
    Leader = 'leader',
    Follower = 'follower',
    Learner = 'learner',
    Candidate = 'candidate',
    Shutdown = 'shutdown',
    Unknown = 'unknown',
    }

export const ServerStateSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): ServerState {
        return object;
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: ServerState): any {
        return self;
    }
}

File: z-clients/rust/src/models/server_state.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
pub enum ServerState {
    #[serde(rename = "leader")]
    Leader,
    #[serde(rename = "follower")]
    Follower,
    #[serde(rename = "learner")]
    Learner,
    #[serde(rename = "candidate")]
    Candidate,
    #[serde(rename = "shutdown")]
    Shutdown,
    #[serde(rename = "unknown")]
    Unknown,
}

impl fmt::Display for ServerState {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let value = match self {
            Self::Leader => "leader",
            Self::Follower => "follower",
            Self::Learner => "learner",
            Self::Candidate => "candidate",
            Self::Shutdown => "shutdown",
            Self::Unknown => "unknown",
        };
        f.write_str(value)
    }
}
